rmf_websocket/test/test_client.cpp

```diff
@@ -94,22 +97,27 @@ TEST_CASE("Client", "Reconnecting server") {
 
 
   jsonString["test"] = "2";
-  broadcaster->publish(jsonString);
-  t2.join();
+  auto tries = 0;
 
-  // This is a horrible piece of code and defeats
-  // the purpose of the tests. But unfortunately websocketpp
-  // does not correctly return if a send was successful or not.
-  // Thus packets may be lost.
-  if (!timed_out)
+  using namespace std::chrono_literals;
+  while (tries < 10 && !terminate_server)
   {
-    REQUIRE(num_msgs == 2);
-
-    REQUIRE(msgs[0]["test"] == "1");
-    REQUIRE(msgs[1]["test"] == "2");
-  }
-  else
-  {
-    std::cerr << "Test timed out" << std::endl;
+    /// TODO(arjoc): Make timeout reconfigureable, that
+    /// way we dont need to wait so long.
+    broadcaster->publish(jsonString);
+    std::this_thread::sleep_for(1000ms);
+    ++tries;
   }
+  t2.join();
+
+// This is a horrible piece of code and defeats
+// the purpose of the tests. But unfortunately websocketpp
+// does not correctly return if a send was successful or not.
+// Thus packets may be lost.
```

Comment:
I believe this comment is no longer true. The test now is not ideal but we no longer close an eye and make the test pass on time out.